server/src/main/java/io/crate/execution/ddl/TransportSchemaUpdateAction.java

```diff
@@ -302,16 +302,19 @@ private static void populateColumnPositions(Map<String, Object> mapping,
             contentPath.add(name);
             Map<String, Object> columnProperties = (Map<String, Object>) e.getValue();
             columnProperties = furtherColumnProperties(columnProperties);
+            assert columnProperties.containsKey("position") && columnProperties.get("position") != null : "Column position is missing: " + name;
+            // BWC compatibility with nodes < 5.1, position could be NULL if column is created on that nodes
             Integer position = (Integer) columnProperties.get("position");
-            assert position != null : "position should not be null";
-            if (position < 0) {
-                columnPositionResolver.addColumnToReposition(contentPath.pathAsText(""),
-                                                             position,
-                                                             columnProperties,
-                                                             (cp, p) -> cp.put("position", p),
-                                                             contentPath.currentDepth());
-            } else {
-                maxColumnPosition[0] = Math.max(maxColumnPosition[0], position);
+            if (position != null) {
```

Comment:
Afaik a negative position value indicates that the position must be re-calculated, isn't this true?
If so, why don't we treat a `NULL` value in the same way?